### PR TITLE
fix the SurfaceView not hide issue.

### PR DESCRIPTION
**Pull request by guoguo338**
This PR fixed the issue: https://github.com/flutter/flutter/issues/113601.

The reason is that the embedded PlatformView holds a SurfaceView which needs force to set View.GONE or View.VISIBLE when hide/show.

Below is the demo code:
image

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/engine/mutatorsstack/FlutterMutatorView.java [hunk 1]
```diff
@@ -178,6 +179,14 @@ public boolean requestSendAccessibilityEvent(View child, AccessibilityEvent even
     return super.requestSendAccessibilityEvent(child, event);
   }
 
+  @Override
+  public void setVisibility(int visibility) {
+    final View childView = this.getChildAt(0);
+    if (childView != null && childView instanceof SurfaceView) {
+      childView.setVisibility(visibility);
+    }
```
Comment (cyanglaz) [inline shell/platform/android/io/flutter/embedding/engine/mutatorsstack/FlutterMutatorView.java:187]: Don't you need to call super?

**Review by cyanglaz — COMMENTED** [review]
Thanks for the fix! The PR needs tests in order to be landed.

**Comment by guoguo338** [comment]
@cyanglaz , Could you help to take a further review?

**Comment by cyanglaz** [comment]
I'm not familiar with SurfaceView API so this is unclear to me why SurfaceView needs to be explicitly set to gone or visible. Nothing is mentioned in https://developer.android.com/reference/android/view/SurfaceView. Could you explain more? Would be nice if you can add a comment in the code to explain it.

@reidbaker Might have more thoughts.

**Comment by guoguo338** [comment]
> I'm not familiar with SurfaceView API so this is unclear to me why SurfaceView needs to be explicitly set to gone or visible. Nothing is mentioned in https://developer.android.com/reference/android/view/SurfaceView. Could you explain more? Would be nice if you can add a comment in the code to explain it.
> 
> @reidbaker Might have more thoughts.

@reidbaker ，friendly ping. :-)

**Comment by stuartmorgan** [comment]
> You can check this issue: [flutter/flutter#105203](https://github.com/flutter/flutter/issues/105203).

I don't see an answer to the question there. Normally, when setting a view's visibility, you don't need to set the visibility of subviews as well. Why is `SurfaceView` different? There needs to be a comment in the new code explaining why it's a special case.

> For FlutterSurfaceView, setting View.GONE to the current FlutterView will not take effect since it is not in the view tree.

How can the child view be in the view tree if the parent view isn't?

**Comment by chinmaygarde** [comment]
@guoguo338 Any updates to @stuartmorgan queries?

**Comment by GaryQian** [comment]
@guoguo338 Let us know if you had a chance to consider @stuartmorgan 's questions

**Comment by gmackall** [comment]
Thank you for your contribution. I'm going to close this PR for now since there are outstanding comments, just to get this off our PR review queue. Please don't hesitate to submit a new PR if you have the time to address the review comments. Thanks!

**Pull request closed without merging**